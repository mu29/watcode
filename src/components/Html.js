/* eslint-disable react/no-danger */
import React from 'react'
import { PropTypes } from 'prop-types'
import { Helmet } from 'react-helmet'

const Html = ({ styles, state, children }) => {
  const helmet = Helmet.renderStatic()
  const htmlAttrs = helmet.htmlAttributes.toComponent()
  const bodyAttrs = helmet.bodyAttributes.toComponent()

  return (
    <html lang="ko" { ...htmlAttrs }>
      <head>
        { helmet.title.toComponent() }
        { helmet.meta.toComponent() }
        { helmet.link.toComponent() }
        <meta charSet="utf8" />
        <link href="/style.css" rel="stylesheet" type="text/css" />
        <link href="//spoqa.github.io/spoqa-han-sans/css/SpoqaHanSans-kr.css" rel="stylesheet" type="text/css" />
        <script defer src="https://use.fontawesome.com/releases/v5.0.13/js/all.js" integrity="sha384-xymdQtn1n3lH2wcu0qhcdaOpQwyoarkgLVxC/wZ5q7h9gHtxICrpcaSUfygqZGOe" crossOrigin="anonymous" />
        { styles }
      </head>
      <body { ...bodyAttrs }>
        <main id="app">{ children }</main>
        <script dangerouslySetInnerHTML={{ __html: state }} />
        <script src="/client.js" />
      </body>
    </html>
  )
}

Html.propTypes = {
  styles: PropTypes.node.isRequired,
  state: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default Html
