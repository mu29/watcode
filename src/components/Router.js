import React from 'react'
import { PropTypes } from 'prop-types'
import { Switch, Route, Redirect } from 'react-router-dom'
import routes from 'routes'

export default class Router extends React.Component {
  static propTypes = {
    isLoggedIn: PropTypes.bool,
  }

  static defaultProps = {
    isLoggedIn: false,
  }

  renderComponent = (route, isLoggedIn) => (props) => {
    if (!isLoggedIn && route.private) {
      return <Redirect to={{ pathname: '/login', state: { from: props.location } }} />
    }

    if (isLoggedIn && ['/login', '/'].includes(route.path)) {
      return <Redirect to="/profile" />
    }

    return <route.component { ...props } />
  }

  renderRoute = route => (
    <Route
      key={ route.path }
      path={ route.path }
      exact={ route.exact }
      render={ this.renderComponent(route, this.props.isLoggedIn) }
    />
  )

  render() {
    return (
      <Switch>
        { routes.map(route => this.renderRoute(route)) }
      </Switch>
    )
  }
}
