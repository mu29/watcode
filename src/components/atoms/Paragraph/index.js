import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import { color, fontFamily, fontSize } from 'services/style'
import { nl2br } from 'services/utils'

const styles = css`
  color: ${color};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  line-height: 1.4;
`

const Paragraph = styled(({
  children, ...props
}) => React.createElement('p', props, nl2br(children)))`${styles}`

Paragraph.propTypes = {
  color: PropTypes.string,
  font: PropTypes.string,
  fontSize: PropTypes.number,
}

Paragraph.defaultProps = {
  color: 'gray.90',
  font: 'default',
  fontSize: 16,
}

export default Paragraph
