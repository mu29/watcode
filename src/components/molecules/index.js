export { default as Field } from './Field'
export { default as IconButton } from './IconButton'
export { default as Tab } from './Tab'
