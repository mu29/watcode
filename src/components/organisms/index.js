export { default as ArtworkList } from './ArtworkList'
export { default as Header } from './Header'
export { default as LoginForm } from './LoginForm'
export { default as ReduxField } from './ReduxField'
export { default as RegisterForm } from './RegisterForm'
export { default as SearchBar } from './SearchBar'
