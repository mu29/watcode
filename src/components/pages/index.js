export { default as HomePage } from './HomePage'
export { default as RankingPage } from './RankingPage'
export { default as BookmarkPage } from './BookmarkPage'
