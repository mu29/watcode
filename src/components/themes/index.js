export { default as basicTheme } from './basic'
