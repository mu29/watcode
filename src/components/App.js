import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { injectGlobal, ThemeProvider } from 'styled-components'
import { authorizeActions, readBookmarksActions } from 'store/actions'
import { auth } from 'services/firebase'
import { basicTheme } from './themes'
import Router from './Router'

/* eslint-disable no-unused-expressions */
injectGlobal`
  * {
    box-sizing: border-box;
  }
  body {
    margin: 0;
  }
`

export default class App extends React.Component {
  static propTypes = {
    store: PropTypes.shape({
      dispatch: PropTypes.func,
      getState: PropTypes.func,
    }).isRequired,
  }

  componentWillMount() {
    const { store } = this.props
    auth.onAuthStateChanged((user) => {
      store.dispatch(authorizeActions.request({ user }))
      store.dispatch(readBookmarksActions.request())
    })
  }

  render() {
    const state = this.props.store.getState()
    const isLoggedIn = state.auth.user.email !== undefined
    return (
      <React.Fragment>
        <Helmet titleTemplate="watcode">
          <title>왓코드 - 오늘 볼 만화 추천해줘!</title>
          <meta charSet="UTF-8" />
          <meta name="description" content="" />
          <meta name="viewport" content="width=device-width, initial-scale=1.0" />
          <meta property="og:site_name" content="watcode" />
        </Helmet>
        <ThemeProvider theme={ basicTheme }>
          <Router isLoggedIn={ isLoggedIn } />
        </ThemeProvider>
      </React.Fragment>
    )
  }
}
