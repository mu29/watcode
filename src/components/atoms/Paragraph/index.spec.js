import React from 'react'
import { shallow } from 'enzyme'
import { basicTheme } from 'components/themes'
import 'jest-styled-components'
import Paragraph from '.'

const wrap = (props = {}) => shallow(<Paragraph theme={ basicTheme } { ...props } />).dive()

describe('<Paragraph />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap()
    expect(wrapper).toMatchSnapshot()
  })

  it('자식 컴포넌트가 들어온 경우 그려준다', () => {
    const wrapper = wrap({ children: 'test' })
    expect(wrapper).toMatchSnapshot()
  })

  it('임의의 속성이 들어온 경우 그려준다', () => {
    const wrapper = wrap({ name: 'foo' })
    expect(wrapper.props().name).toEqual('foo')
  })

  it('기본으로 p 태그를 그려준다', () => {
    const wrapper = wrap()
    expect(wrapper.find('p')).toHaveLength(1)
  })
})
