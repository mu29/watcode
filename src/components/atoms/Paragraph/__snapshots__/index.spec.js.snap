// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> 자식 컴포넌트가 들어온 경우 그려준다 1`] = `
.c0 {
  color: #424242;
  font-family: 'Spoqa Han Sans','Sans-serif','Helvetica Neue','Nanum Gothic';
  font-size: 1rem;
  line-height: 1.4;
}

<p
  className="c0"
  color="gray.90"
  font="default"
  fontSize={16}
  theme={
    Object {
      "fonts": Object {
        "default": "'Spoqa Han Sans', 'Sans-serif', 'Helvetica Neue', 'Nanum Gothic'",
        "monospace": "'Monaco', 'Consolas'",
      },
      "palette": Object {
        "gray": Object {
          "10": "#FAFAFA",
          "100": "#2D2D2D",
          "20": "#F5F5F5",
          "30": "#EEEEEE",
          "40": "#E0E0E0",
          "50": "#BDBDBD",
          "60": "#9E9E9E",
          "70": "#757575",
          "80": "#616161",
          "90": "#424242",
        },
        "red": Object {
          "default": "#FF8787",
        },
        "transparent": Object {
          "default": "transparent",
        },
        "white": Object {
          "default": "#FFF",
        },
        "yellow": Object {
          "dark": "#C49000",
          "default": "#FBC02D",
        },
      },
      "sizes": Object {},
    }
  }
>
  test
</p>
`;

exports[`<Paragraph /> 정상적으로 렌더링된다 1`] = `
.c0 {
  color: #424242;
  font-family: 'Spoqa Han Sans','Sans-serif','Helvetica Neue','Nanum Gothic';
  font-size: 1rem;
  line-height: 1.4;
}

<p
  class="c0"
  color="gray.90"
  font="default"
  font-size="16"
  theme="[object Object]"
/>
`;
